synthesizer/process/src/stack/helpers/synthesize.rs

```diff
@@ -32,6 +32,11 @@ impl<N: Network> Stack<N> {
         let program_id = self.program_id();
         // Retrieve the function input types.
         let input_types = self.get_function(function_name)?.input_types();
+        // Retrieve the program checksum, if the program has a constructor.
+        let program_checksum = match self.program().contains_constructor() {
+            true => Some(self.program_checksum_as_field()?),
+            false => None,
+        };
```

Comment:
I'm pretty sure I've seen this a few times by now, perhaps it could be turned into a method?